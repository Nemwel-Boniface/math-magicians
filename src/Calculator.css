
.left {
  position: absolute;
  left: 20%;
  top: 50%;
}
.left h1 {
  font-size: 45px;
}
.right {
  width: 400px;
  position: absolute;
  right: 1%;
  top: 60%;
  transform: translate(-50%, -50%);
  box-shadow: 0 0 25px 0 #aaa;
}

.resultsPanel {
  background-color: #858694;
  padding: 30px;
  text-align: right;
  color: #fff;
}

.buttons button {
  width: 100px;
  padding: 20px;
}

.buttons .double {
  width: 200px;
}

.buttons .x {
  background: linear-gradient(to bottom, rgba(252, 156, 23, 1) 0%, rgba(247, 126, 27, 1) 100%);
  transition: 0.5s;
}

.buttons .x:hover {
  background: linear-gradient(to bottom, rgba(252, 156, 23, 0.7) 0%, rgba(247, 126, 27, 0.7) 100%);
  color: #fff;
}
