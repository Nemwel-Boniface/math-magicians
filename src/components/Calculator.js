import React, { useState } from 'react';
import calculate from '../logic/calculate';

const Calculator = () => {
  const [state, setState] = useState({ total: 0, next: null, operation: null });
  const { total } = state;
  const { next } = state;
  const { operation } = state;

  const handleCalc = (e) => {
    setState((state) => calculate(state, e.target.textContent));
  };

  return (
    <>
      <div className="calculator">
        <div className="resultsPanel">
          <span className="result">
            {total}
            {operation}
            {next}
          </span>
        </div>
        <div className="buttons">
          <button onClick={handleCalc} type="button" className="btn1">AC</button>
          <button onClick={handleCalc} type="button" className="btn2">+/-</button>
          <button onClick={handleCalc} type="button" className="btn3">%</button>
          <button onClick={handleCalc} type="button" className="btn4 x">รท</button>
          <button onClick={handleCalc} type="button" className="btn5">7</button>
          <button onClick={handleCalc} type="button" className="btn6">8</button>
          <button onClick={handleCalc} type="button" className="btn7">9</button>
          <button onClick={handleCalc} type="button" className="btn8 x">x</button>
          <button onClick={handleCalc} type="button" className="btn9">4</button>
          <button onClick={handleCalc} type="button" className="btn0">5</button>
          <button onClick={handleCalc} type="button" className="btn11">6</button>
          <button onClick={handleCalc} type="button" className="btn12 x">-</button>
          <button onClick={handleCalc} type="button" className="btn13">1</button>
          <button onClick={handleCalc} type="button" className="btn14">2</button>
          <button onClick={handleCalc} type="button" className="btn15">3</button>
          <button onClick={handleCalc} type="button" className="btn16 x">+</button>
          <button onClick={handleCalc} type="button" className="btn17 double">0</button>
          <button onClick={handleCalc} type="button" className="btn18">.</button>
          <button onClick={handleCalc} type="button" className="btn19 x">=</button>
        </div>
      </div>
    </>
  );
};

export default Calculator;
