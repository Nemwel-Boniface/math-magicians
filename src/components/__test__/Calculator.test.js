import React from "react";
import { render, cleanup } from "@testing-library/react";
import Calculator from '../Calculator';

afterEach(cleanup)
it('Should create a calculator snapshot', () => {
  const { asFragment } = render(<Calculator />)
  expect(asFragment(<Calculator />)).toMatchSnapshot()
})