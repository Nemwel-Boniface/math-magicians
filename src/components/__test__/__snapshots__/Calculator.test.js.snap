// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should create a calculator snapshot 1`] = `
<DocumentFragment>
  <div
    class="calculator w3-animate-zoom"
  >
    <div
      class="left"
    >
      <h1>
        Let's do some math!
      </h1>
    </div>
    <div
      class="right"
    >
      <div
        class="resultsPanel"
      >
        <span
          class="result"
        >
          0
        </span>
      </div>
      <div
        class="buttons"
      >
        <button
          class="btn1"
          type="button"
        >
          AC
        </button>
        <button
          class="btn2"
          type="button"
        >
          +/-
        </button>
        <button
          class="btn3"
          type="button"
        >
          %
        </button>
        <button
          class="btn4 x"
          type="button"
        >
          รท
        </button>
        <button
          class="btn5"
          type="button"
        >
          7
        </button>
        <button
          class="btn6"
          type="button"
        >
          8
        </button>
        <button
          class="btn7"
          type="button"
        >
          9
        </button>
        <button
          class="btn8 x"
          type="button"
        >
          x
        </button>
        <button
          class="btn9"
          type="button"
        >
          4
        </button>
        <button
          class="btn0"
          type="button"
        >
          5
        </button>
        <button
          class="btn11"
          type="button"
        >
          6
        </button>
        <button
          class="btn12 x"
          type="button"
        >
          -
        </button>
        <button
          class="btn13"
          type="button"
        >
          1
        </button>
        <button
          class="btn14"
          type="button"
        >
          2
        </button>
        <button
          class="btn15"
          type="button"
        >
          3
        </button>
        <button
          class="btn16 x"
          type="button"
        >
          +
        </button>
        <button
          class="btn17 double"
          type="button"
        >
          0
        </button>
        <button
          class="btn18"
          type="button"
        >
          .
        </button>
        <button
          class="btn19 x"
          type="button"
        >
          =
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
